filters/openpolicyagent/opaauthorizerequest/opaauthorizerequest_test.go

```diff
@@ -57,6 +57,34 @@ func TestAuthorizeRequestFilter(t *testing.T) {
 			backendHeaders:    make(http.Header),
 			removeHeaders:     make(http.Header),
 		},
+		{
+			msg:               "Allow Requests with trailing ? in URL",
+			filterName:        "opaAuthorizeRequest",
+			bundleName:        "somebundle.tar.gz",
+			regoQuery:         "envoy/authz/allow",
+			requestPath:       "/allow?",
```

Comment:
Shouldn't this be `allow-with-query?` ?

Reply:
Or rather, I don't this this test is super meaningful as you can also pass `/allow?foo=bar` and it will pass as query parameters are ignored in Rego.

Reply:
> You might want to validate the that path as visible in Reno does not actually have the ? in it, which I think is what you want to make sure of.

yes, that was my idea when I suggested this test case.

Reply:
> Or rather, I don't think this test is super meaningful as you can also pass `/allow?foo=bar` and it will pass as query parameters are ignored in Rego. You might want to validate the that path as visible in Reno does not actually have the `?` in it, which I think is what you want to make sure of...

My thinking was that if the client requests `/allow?` (without queries) and the policy only allows only path `/allow` (and does not take into account query params), the test would assert that path passed to rego does not have the consider trailing `?`

Won't this cover what we want to check?

The other possible test case i thought could cover the scenario explicitly was,
```
		{
			msg:               "OPA does not consider trailing ? in URL",
			filterName:        "opaAuthorizeRequest",
			bundleName:        "somebundle.tar.gz",
			regoQuery:         "envoy/authz/allow",
			requestPath:       "/ends-with?",
			requestMethod:     "GET",
			contextExtensions: "",
			expectedStatus:    http.StatusForbidden,
			expectedHeaders:   make(http.Header),
			backendHeaders:    make(http.Header),
			removeHeaders:     make(http.Header),
		},
		
		allow {
			input.parsed_path = [ "ends-with?" ]
		}
```

Reply:
> and the policy only allows only path /allow (and does not take into account query params)

I think I made a confusion (sorry about that). My idea was to test with the trailing `?` (with empty query params) to see and document the behaviour of rego with this type of paths.

I was trying to answer this questions
* will the path be `/allow?` or `/allow`?
* If for some reason the rego role allow any query param will this also be a valid query param or not?

Reply:
@mefarazath I think the added test case makes sense. I'd also add a condition in Rego to check that the `parsed_query` is empty. This will at least document the current behaviour.

I am curious to see if this looks like this in Rego
`input.parsed_path == [ "ends-with?" ]`
or
`input.parsed_path == [ "ends-with" ]`

It should actually be the latter, stripping the question mark.

Reply:
`?` should not be part of the "path".